Fighters.cpp

```diff
@@ -1,31 +1,38 @@
 #include <iostream>
+#include <string>
 #include "Fighters.h"
 
-Fighters::Fighters(std::string ID,int hp, int dmg) : ID(ID), hp(hp), dmg(dmg) 
-{}
 
-std::string Fighters::getID() const
+void Fighters::Fight(Fighters& enemy)
 {
-    return ID;
+    while (hp > 0)
+    {
+        (enemy.getHP()-dmg > 0) ? enemy.hp -= getDmg() : enemy.hp = 0;
+		if (enemy.getHP() == 0) { break; }
+    	(hp - enemy.getDmg() > 0) ? hp -= enemy.dmg : hp = 0;
+    }
 }
-
-int Fighters::getHP() const
+Fighters Fighters::parseUnit(const std::string &jsonfile)
 {
-    return hp;
-}
+    {
+        std::ifstream file("Units/" + jsonfile + ".json");
```

Comment:
Nem javaslom "beledrótozni" a mappaszerkezetet a kódba. Ha a felhasználó egy külön könyvtárban tárolja a unitokat, akkor majd megadja parancssorban a könyvtárat is.